src/pools/PoolRegistry.sol

```diff
@@ -14,13 +15,24 @@ contract PoolRegistry is Auth, IPoolRegistry {
 
     uint48 public latestId;
 
+    mapping(AssetId => uint8) internal _decimals;
+
     mapping(PoolId => bytes) public metadata;
     mapping(PoolId => AssetId) public currency;
     mapping(PoolId => mapping(address => bool)) public isAdmin;
     mapping(PoolId => mapping(bytes32 => address)) public dependency;
 
     constructor(address deployer) Auth(deployer) {}
 
+    /// @inheritdoc IPoolRegistry
+    function registerAsset(AssetId assetId, uint8 decimals_) external auth {
+        require(_decimals[assetId] == 0, AssetAlreadyRegistered());
+
+        _decimals[assetId] = decimals_;
+
+        emit NewAsset(assetId, decimals_);
+    }
```

Comment:
I like the simplification of this PR, but I have just one "disagree" regarding having this pool-unrelated method into the `PoolRegistry`. Given this is global, why not rescue some previous idea from you about have a general contract for general stuff like:

`HubRegistry.setValue(bytes key, uint256 value)` where key here is `keccak("asset", assetId)`? (although not sure about the performance). If not, I think we could have a super small `AssetRegistry` contract where to store this, but having it inside PoolRegistry seems very weird.

Reply:
Another option could be to rename `PoolRegistry` to `HubRegistry`

Reply:
I like the idea of the renaming.

I think we shouldn't add another contract, we just will add even more external calls.